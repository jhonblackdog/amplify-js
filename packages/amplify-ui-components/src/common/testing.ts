export const pixelThreshold = 10;
