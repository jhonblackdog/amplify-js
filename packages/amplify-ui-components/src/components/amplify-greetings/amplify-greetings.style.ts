import { css } from 'emotion';

export const greetings = css`
  position: relative;
  border: 1px solid transparent;
  background-color: var(--section-background-color);
  border-color: var(--light-grey);
  padding: 10px;
  font-family: var(--font-family);
`;