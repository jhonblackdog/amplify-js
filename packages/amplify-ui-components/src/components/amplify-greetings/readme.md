# amplify-greetings



<!-- Auto Generated Below -->


## Properties

| Property                | Attribute  | Description                                  | Type                                                | Default                        |
| ----------------------- | ---------- | -------------------------------------------- | --------------------------------------------------- | ------------------------------ |
| `handleAuthStateChange` | --         | Auth state change handler for this component | `(nextAuthState: AuthState, data?: object) => void` | `dispatchAuthStateChangeEvent` |
| `logo`                  | --         | Logo displayed inside of the header          | `FunctionalComponent<{}>`                           | `null`                         |
| `username`              | `username` | Username displayed in the greetings          | `string`                                            | `null`                         |


## Dependencies

### Depends on

- [amplify-nav](../amplify-nav)
- [amplify-sign-out](../amplify-sign-out)

### Graph
```mermaid
graph TD;
  amplify-greetings --> amplify-nav
  amplify-greetings --> amplify-sign-out
  amplify-sign-out --> amplify-button
  style amplify-greetings fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
