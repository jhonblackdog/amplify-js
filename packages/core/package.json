{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
