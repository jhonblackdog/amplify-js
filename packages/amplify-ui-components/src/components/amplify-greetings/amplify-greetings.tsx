import { Component, h, Prop, FunctionalComponent } from '@stencil/core';
import { AuthStateHandler } from '../../common/types/auth-types';
import { dispatchAuthStateChangeEvent } from '../../common/helpers';
@Component({
  tag: 'amplify-greetings',
  styleUrl: 'amplify-greetings.scss',
  shadow: true,
})
export class AmplifyGreetings {
  /** Username displayed in the greetings */
  @Prop() username: string = null;
  /** Logo displayed inside of the header */
  @Prop() logo: FunctionalComponent | null = null;
  /** Auth state change handler for this component */
  @Prop() handleAuthStateChange: AuthStateHandler = dispatchAuthStateChangeEvent;

  render() {
    return (
      <header class="greetings">
        {// TODO: user <amplify-logo> instead of <span>
        this.logo && <span>{this.logo}</span>}
        <amplify-nav>
          {this.username && (
            <slot name="greetings-message">
              <span>Hello, {this.username}</span>
            </slot>
          )}
          <amplify-sign-out handleAuthStateChange={this.handleAuthStateChange} />
        </amplify-nav>
      </header>
    );
  }
}
