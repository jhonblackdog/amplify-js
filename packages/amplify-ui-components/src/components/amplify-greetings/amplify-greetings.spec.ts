import { newSpecPage } from '@stencil/core/testing';
import { AmplifyGreetings } from './amplify-greetings';

describe('amplify-greetings spec:', () => {
	describe('Component logic ->', () => {
		let greetings;

		beforeEach(() => {
			greetings = new AmplifyGreetings();
		});

		it('should have user prop set to null by default', () => {
			expect(greetings.user).toBe(null);
		});

		it('should have logo prop set to null by default', () => {
			expect(greetings.logo).toBe(null);
		});

		it('should have overrideStyle prop set to false by default', () => {
			expect(greetings.overrideStyle).toBe(false);
		});
	});
	describe('Render logic ->', () => {
		it('should render', async () => {
			const page = await newSpecPage({
				components: [AmplifyGreetings],
				html: `<amplify-greetings></amplify-greetings>`
			});

			expect(page.root).toMatchSnapshot();
		});
	});
});