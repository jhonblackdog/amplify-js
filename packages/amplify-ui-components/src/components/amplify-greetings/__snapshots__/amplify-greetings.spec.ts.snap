// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-greetings spec: Render logic -> should render 1`] = `
<amplify-greetings>
  <mock:shadow-root>
    <header class="greetings">
      <amplify-nav>
        <amplify-sign-out></amplify-sign-out>
      </amplify-nav>
    </header>
  </mock:shadow-root>
</amplify-greetings>
`;
